pkg/cli/bootstrap.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2021 Outreach Corporation. All Rights Reserved.
+
+// Description: This file contains cli functions used in bootstrap
+// and eventually in stencil.
+package cli
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"os/user"
+	"runtime"
+	"runtime/debug"
+	"strings"
+	"syscall"
+
+	"github.com/getoutreach/gobox/pkg/app"
+	"github.com/getoutreach/gobox/pkg/cfg"
+	"github.com/getoutreach/gobox/pkg/exec"
+	"github.com/getoutreach/gobox/pkg/log"
+	"github.com/getoutreach/gobox/pkg/secrets"
+	"github.com/getoutreach/gobox/pkg/trace"
+	"github.com/getoutreach/gobox/pkg/updater"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	// UpdateExitCode is the exit code returned when an update ocurred
+	UpdateExitCode = 5
+)
+
+// overrideConfigLoaders fakes certain parts of the config that usually get pulled
+// in via mechanisms that don't make sense to use in CLIs.
+func overrideConfigLoaders(honeycombApiKey string, dataset string) {
+	var fallbackSecretLookup func(context.Context, string) ([]byte, error)
+	fallbackSecretLookup = secrets.SetDevLookup(func(ctx context.Context, key string) ([]byte, error) {
+		if key == "APIKey" {
+			return []byte(honeycombApiKey), nil
+		}
+
+		return fallbackSecretLookup(ctx, key)
+	})
+
+	log.SetOutput(io.Discard)
+
+	fallbackConfigReader := cfg.DefaultReader()
+	cfg.SetDefaultReader(func(fileName string) ([]byte, error) {
+		if fileName == "trace.yaml" {
+			traceConfig := &trace.Config{
+				Honeycomb: trace.Honeycomb{
+					Enabled: true,
+					APIHost: "https://api.honeycomb.io",
+					APIKey: cfg.Secret{
+						Path: "APIKey",
+					},
+					Dataset:       dataset,
+					SamplePercent: 100,
+				},
+			}
+			b, err := yaml.Marshal(&traceConfig)
+			if err != nil {
+				panic(err)
+			}
+			return b, nil
+		}
+
+		return fallbackConfigReader(fileName)
+	})
+}
+
+// intPtr turns an int into a *int
+func intPtr(i int) *int {
+	return &i
+}
+
+// funcPtr turns a func into a *func
+func funcPtr(fn func()) *func() {
+	return &fn
+}
+
+// urfaveRegisterShutdownHandler registers a signal notifier that translates various term
+// signals into context cancel
+func urfaveRegisterShutdownHandler(cancel context.CancelFunc) {
+	// handle ^C gracefully
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM, syscall.SIGHUP)
+	go func() {
+		<-c
+		cancel()
+	}()
+}
+
+// setupTracer sets up a root trace for the CLI and initializes the tracer
+func setupTracer(ctx context.Context, name string) context.Context {
```

Comment:
maybe modify this function to return a closer function that calls `trace.CloseTracer(ctx)`

Reply:
nvm i saw what you did. kinda sketchy but i understand where it comes from with how generic this code has to be